x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -317,12 +332,11 @@ export const useSourcererDataView = (
       sourcererScope,
     };
   });
-
   const selectedPatterns = useMemo(
     () =>
       scopeSelectedPatterns.some((index) => index === LOGS_WILDCARD_INDEX)
-        ? [...scopeSelectedPatterns, EXCLUDE_ELASTIC_CLOUD_INDEX]
-        : scopeSelectedPatterns,
+        ? [...sortWithExcludesAtEnd(scopeSelectedPatterns), ...EXCLUDE_ELASTIC_CLOUD_INDICES]
```

Comment:
Given that this PR implements https://github.com/elastic/kibana/issues/130753 , such that it's possible to add exclude patterns directly to the `securitySolution:defaultIndex`, would you and @paulewing be willing to consider the following changes?

- Remove this code, which automatically appends a hard-coded list of indices to exclude
- Relocate the hard-coded excludes to the default value of the `securitySolution:defaultIndex`
- Remove the help text shown in the query inspector, which appears to be inconsistently used today

There are a few reasons why the proposal above may (now) be a better option:

1) It's possible to use the `Security Data View`, which is synchronized with the `securitySolution:defaultIndex` setting, in other parts of Kibana, like Discover. However today the data shown in Discover and the Security solution will be subtilely different, because, for example, the hard-coded `-*elastic-cloud-logs-*` exclude pattern won't be applied to Discover.

2) Within the Security Solution, it's effectively impossible to add any of the indexes that today are automatically excluded. For example, explicitly adding `*elastic-cloud-logs-*` to the `securitySolution:defaultIndex` setting will have no effect, because the code here will (only within the Security Solution) ensure it's excluded anyway.

3) The `securitySolution:defaultIndex` setting would be [WYSIWYG](https://en.wikipedia.org/wiki/WYSIWYG). As a result, the help text in the query inspector, which today appears to be inconsistently used, can be eliminated.

For example, the Hosts > Events query inspector, does NOT include the help text, per the screenshot below:

![inspect_host_events](https://user-images.githubusercontent.com/4459398/166083853-2dadb677-c8a0-40e2-b804-b50ae6e91487.png)

_Above: Inspecting the Hosts > Events query inspector shows the hard-coded exclude patterns were  automatically applied, but it does NOT include help text_

In contrast, the Timeline query inspector includes the following help text:

```
When the logs-* index pattern is selected, Elastic cloud logs are exclude from the search.
```

per the screenshot below:

![inspect-timeline](https://user-images.githubusercontent.com/4459398/166083995-9b5d3b69-2e54-44eb-b1e7-7ec066d48f73.png)

_Above: Inspecting the Timeline query reveals that the exclude patterns were automatically added, AND it includes the help text explaining them_

On caveat to the proposal above is Elastic Cloud users who already overrode the default  `securitySolution:defaultIndex` setting may, after upgrading, start to see Elastic cloud logs in the security solution until they manually add the exclude pattern.

Regardless of whether or not the suggestion above is implemented, this PR should be included in the release notes.